api4/cloud.go

```diff
@@ -767,3 +769,28 @@ func handleCheckCWSConnection(c *Context, w http.ResponseWriter, r *http.Request
 
 	ReturnStatusOK(w)
 }
+
+func selfServeDeleteWorkspace(c *Context, w http.ResponseWriter, r *http.Request) {
+	bodyBytes, err := io.ReadAll(r.Body)
+	if err != nil {
+		c.Err = model.NewAppError("Api4.selfServeDeleteWorkspace", "api.cloud.app_error", nil, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	defer r.Body.Close()
+
+	var deleteRequest *model.WorkspaceDeletionRequest
+	if err = json.Unmarshal(bodyBytes, &deleteRequest); err != nil {
+		c.Err = model.NewAppError("Api4.selfServeDeleteWorkspace", "api.cloud.app_error", nil, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	if err := c.App.Cloud().SelfServeDeleteWorkspace(c.AppContext.Session().UserId, deleteRequest); err != nil {
+		c.Err = model.NewAppError("Api4.selfServeDeleteWorkspace", "api.server.cws.delete_workspace.app_error", nil, "CWS Server failed to delete workspace.", http.StatusInternalServerError)
+		return
+	}
+
+	c.App.Srv().GetTelemetryService().SendTelemetry("delete_workspace_feedback", deleteRequest.Feedback.ToMap())
+
+	ReturnStatusOK(w)
```

Comment:
From a user experience POV, deletions should be immediate, even if we hold onto their workspace data for 7 days.
Should we invalidate all sessions somewhere in this handler, if at all possible?

Took a quick look and it looks like the session store has the functionality to do this with `RemoveAllSessions`.